spec/haskell/src/SEL4/Object/ObjectType/AARCH64.hs

```diff
@@ -304,3 +304,12 @@ prepareThreadDelete thread = do
       Just ptr -> dissociateVCPUTCB ptr thread
       _ -> return ()
     fpuRelease thread
+
+-- Save and clear FPU and VCPU state before setting the domain of a TCB, to ensure that
```

Comment:
This was confusing to read. First I thought "what, any TCB?", then I thought "only the current TCB?", then I thought "hang on, it only does if if we change the TCB, not when we set it". It seems it's something more like "Save/clear any FPU and VCPU state of a TCB before changing its domain, [...]"? (same comment for other arches)

Reply:
(I recognise the original came from other commits, but since we're here...)